### ISSUE TITLE
For text swatches "getSwatchesByOptionsId()" method always returns admin store swatches - not the current store ones.

### ISSUE THREAD

**Issue by Mobecls** [magento/magento2#19072]
### Preconditions (*)

1.Magento 2.2.6 and Magento 2.2.5
2.PHP7.0 and PHP7.1

### Steps to reproduce (*)

1.Install Magento 2.2.6 (or Magento 2.2.5 - bug exists in both versions).
2.Create attribute with "Text Swatch" input type.
3.Add one or more swatches to that attribute, making sure that swatches in "Admin" column were different from swatches in "Default Store View". On the following screenshot I showed (as an example) attribute "testattribute" with two swatches added: https://prnt.sc/lery4a
4.Use getSwatchesByOptionsId() method to obtain swatches of this attribute.
To make this step easier, here is a small CLI script that should be run from the root magento directory. When run, this script will show the result of the getSwatchesByOptionsId() method call appied to the "testattribute" attribute options (of course, "testattribute" attribute should be added beforehands as showed at the screenshot from the step 3).

```
<?php
const ATTRIBUTE_CODE = 'testattribute';

use Magento\Framework\App\Bootstrap;
require __DIR__ . '/app/bootstrap.php';

$bootstrap = Bootstrap::create(BP, $_SERVER);

$objectManager = $bootstrap->getObjectManager();

$state = $objectManager->get('Magento\Framework\App\State');
$state->setAreaCode('frontend');

$storeManager = $objectManager->get('Magento\Store\Model\StoreManagerInterface');
$currentStoreId = $storeManager->getStore()->getId();

echo "Current store ID is $currentStoreId" . PHP_EOL;

$swatchesHelper = $objectManager->get('Magento\Swatches\Helper\Data');
$eavConfig = $objectManager->get('Magento\Eav\Model\Config');

$attribute = $eavConfig->getAttribute('catalog_product', ATTRIBUTE_CODE);
$options = $attribute->getSource()->getAllOptions();

$optionIds = array_map(
    function($nextOption) {
        return $nextOption['value'];
    }, $options);

$swatches = $swatchesHelper->getSwatchesByOptionsId($optionIds);
echo PHP_EOL . 'getSwatchesByOptionsId() results:' . PHP_EOL;
print_r($swatches);

```

### Expected result (*)

1. We should see text swatches as they were entered in the "Default Store View" column (see screenshot mentioned at the third step above). Something like this:

```
Current store ID is 1

getSwatchesByOptionsId() results:
Array
(
    [477] => Array
        (
            [swatch_id] => 255
            [option_id] => 477
            [store_id] => 1
            [type] => 0
            [value] => storeSw1
        )

    [478] => Array
        (
            [swatch_id] => 258
            [option_id] => 478
            [store_id] => 1
            [type] => 0
            [value] => storeSw2
        )

)

```
In fact, on Magento 2.2.2 this is the case.

### Actual result (*)

1. Alas, on Magento 2.2.6 and Magento 2.2.5 script's result would be as following:

```
Current store ID is 1

getSwatchesByOptionsId() results:
Array
(
    [4] => Array
        (
            [swatch_id] => 1
            [option_id] => 4
            [store_id] => 0
            [type] => 0
            [value] => admSw1
        )

    [5] => Array
        (
            [swatch_id] => 3
            [option_id] => 5
            [store_id] => 0
            [type] => 0
            [value] => admSw2
        )

)

```

As we can see, getSwatchesByOptionsId() method returned text swatched from the admin store - not the text swatches from the current store.

**Comment by magento-engcom-team** [comment]
Hi @Mobecls. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@Mobecls do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @Mobecls thank you for you report this issue has already fixed in 2.2-develop branch.
![selection_193](https://user-images.githubusercontent.com/40756439/48058839-2f663d00-e1c0-11e8-8d1d-5cdad398f649.png)
